sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala

```diff
@@ -45,7 +45,8 @@ object ExtractPythonUDFFromAggregate extends Rule[LogicalPlan] {
 
   private def hasPythonUdfOverAggregate(expr: Expression, agg: Aggregate): Boolean = {
     expr.find {
-      e => PythonUDF.isScalarPythonUDF(e) && e.find(belongAggregate(_, agg)).isDefined
+      e => PythonUDF.isScalarPythonUDF(e) &&
+        (e.references.isEmpty || e.find(belongAggregate(_, agg)).isDefined)
```

Comment:
I just want to consider some literal inputs like `df2 = df.distinct().withColumn("a", f_udf(f.lit("2")))`.

Reply:
I think `references` is more correct - If we use `children` and then we could have an expression, for example, a literal, we would not try this extraction, ending up with leaving `HashAggregateExec` still binding to the Python UDF because I think its `children` is non-empty but it doesn't belong to the aggregate expression in this case whereas `references` remains empty.

Reply:
Sorry, I wrote a duplicate comment and removed it back. It didn't show up when I write ..

Reply:
Oh, I see, sounds good. Thanks!